### DISCUSSION TITLE
Weblate Postgres Migration

### DISCUSSION THREAD

**Discussion by AKIvan** [WeblateOrg/weblate#11499] [General] [answered]
Hi all, 

I try to migrate old weblate to new one. I take the DB and restore it to new one - this part is successfully done. I have done this couple of times and no problems. 

Now for unknown reason im trying to repeat the steps and im getting the following error when i try to bring weblate up 

```
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0001_initial, 0005_invitation_full_name_invitation_username in weblate_auth).	[6df3f766265942e99c3262dec2a03dc7](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/weblate/services/weblate-service/tasks/6df3f766265942e99c3262dec2a03dc7?region=us-east-1)	Weblate
April 25, 2024 at 23:53 (UTC+2:00)	To fix them run 'python manage.py makemigrations --merge'.
```

Im doing this on RDS Postgres and weblate on ECS. 
The process as i mention it was working couple of times.
1. Create RDS 
2. Make the migration using the 'database.sql' file using the psql command
3. Bring the new weblate up and running - using CDK Code

I try running the following commands 

entryPoint: ["/bin/sh", "-c"],
      command: [
        "weblate migrate --noinput && weblate makemigrations --merge --noinput && weblate runserver"
      ],
Still without success. 
Any ideas or hints how and what to do ?

**Comment by nijel** [comment]
`weblate makemigrations --merge --noinput` is definitely not something you should do, that creates your own migrations which will lead to conflicts you see. Furthermore, this command is supposed to fail, do you patch Weblate somehow?

**Reply by AKIvan** [reply]
No i have not patched and im seeing this error even without the `weblate makemigrations --merge --noinput `
and im just following up with the command from the logs to fix it.
`April 25, 2024 at 23:53 (UTC+2:00)	To fix them run 'python manage.py makemigrations --merge'.`

**Reply by AKIvan** [reply]
Just as an update im using the latest weblate image just point there. 
Before i start the weblate im doing the sql preparation - `psql -h <RDS_Endpoint> -U dbadmin -d weblate -f database.sql` 
and once the DB is done im starting the ECS with the Weblate. I have done this 4 times till now and it was working, im not sure why this is not working now. - Same weblate same DB just added one additional project.

**Reply by nijel** [reply, marked as answer]
Didn't you upgrade across major release? These upgrades are not supported and it might cause weird errors in migrations.

If this is not the case, you probably created a migration that is not part of Weblate, that's why you get this error and suggestion to merge. Do you happen to have code for it around?
